### PR TITLE
DM-31064: Create metric to capture the total number of fake sources used

### PR DESCRIPTION
**Pull request by morriscb**

### CODE REVIEW

File: python/lsst/ap/pipe/metrics.py [hunk 1]
```diff
@@ -139,3 +140,57 @@ def run(self, matchedFakes, band):
...
+    _DefaultName = "apFakesCount"
+    ConfigClass = ApFakesCountMetricConfig
+
+    def run(self, matchedFakes, band):
+        """Compute the completeness of recovered fakes within a magnitude
+        range.
+
+        Parameters
+        ----------
+        matchedFakes : `lsst.afw.table.SourceCatalog` or `None`
```
Comment (isullivan) [inline python/lsst/ap/pipe/metrics.py:176]: Please add a Parameters docstring for `band`

File: python/lsst/ap/pipe/metrics.py [hunk 2]
```diff
@@ -139,3 +140,57 @@ def run(self, matchedFakes, band):
...
+                the ratio (`lsst.verify.Measurement` or `None`)
+        """
+        if matchedFakes is not None:
+            magnitudes = matchedFakes[f"{self.config.magVar}" % band]
+            magCutFakes = matchedFakes[np.logical_and(magnitudes > self.config.magMin,
+                                                      magnitudes < self.config.magMax)]
+            meas = Measurement(self.config.metricName,
+                               len(magCutFakes) * u.count)
+        else:
+            self.log.info("Nothing to do: no matched catalog found.")
```
Comment (isullivan) [inline python/lsst/ap/pipe/metrics.py:194]: Do you expect this task to be run on data without a matched catalog? If it is not expected that `matchedFakes` would be `None`, then this should at least be `self.log.warn`.

Comment (isullivan) [reply python/lsst/ap/pipe/metrics.py:194]: I might suggest slightly changing the wording, as well, to reflect that this task does not try to load a catalog itself: "Nothing to do: no matched catalog supplied."

**Review by isullivan — APPROVED** [review]
Looks good; a few documentation suggestions.

File: python/lsst/ap/pipe/metrics.py (hunk 2 above)
Comment (morriscb) [reply python/lsst/ap/pipe/metrics.py:194]: I'm okay with this being info. It's consistent with the other metrics we current have.

**Pull request merged**